src/modules/Packages.rb

```diff
@@ -2623,6 +2625,45 @@ def report_missing_pattern(pattern_name)
       end
     end
 
+    # Search for providers for a list of tags
+    #
+    # The use case of this method is to convert and array of tags into an array
+    # of packages. If a tag does not have a provider, then the tag is included
+    # in the array.
+    #
+    # @param tags [Array<String>] List of tags (ie. package names) to search for.
+    # @return     [Array<String>] List contaning a package for each tag.
+    # @see find_provider
+    def find_providers(tags)
+      tags.each_with_object([]) do |tag, providers|
+        provider = find_provider(tag)
+        if provider.nil?
+          log.error "Provider not found for '#{tag}'"
+          providers << tag
+        else
+          providers << provider
+        end
+      end
+    end
+
+    # Search a provider for a tag
+    #
+    # If a provider is not found, an error will be logged.
+    #
+    # @param tag [String]     Tag to search a package for.
+    # @return    [String,nil] Name of the package which provides that tag.
+    #                         It returns nil if no provider is found.
+    # @see find_providers
+    def find_provider(tag)
+      providers = Pkg.PkgQueryProvides(tag).select { |provide| provide[1] != :NONE }
+      names = providers.map(&:first)
+      if names.size > 1
+        log.info "More than one provider was found for '#{tag}': "\
```

Comment:
Yes, this looks better :wink: 

Nit picking: I'd use `log.warn` as this might potentially cause some unexpected behavior...